src/gui/shell.cpp

```diff
@@ -750,8 +750,12 @@ void Shell::handleNeovimNotification(const QByteArray &name, const QVariantList&
 		if (guiEvName == "Font") {
 			handleGuiFontFunction(args);
 		} else if (guiEvName == "Foreground" && args.size() == 1) {
-			activateWindow();
-			raise();
+			if (isWindow()) {
+				activateWindow();
+				raise();
```

Comment:
Should we use the same approach as in `MainWindow::neovimForeground()`?

Reply:
I updated this to use the same approach in both.

Reply:
Thanks!